src/vdaf.rs

```diff
@@ -47,6 +49,11 @@ pub enum VdafError {
     /// failure when calling getrandom().
     #[error("getrandom: {0}")]
     GetRandom(#[from] getrandom::Error),
+
+    /// IDPF error.
+    #[cfg(feature = "experimental")]
+    #[error("idpf error: {0}")]
+    Idpf(#[from] IdpfError),
```

Comment:
I think we have a problem if we need to modify generic interfaces exported from `mod vdaf` to accommodate new VDAFs. Our goal is that someone should be able to implement new VDAFs outside of crate `prio` by implementing traits `Vdaf, Aggregator, Collector, Client`. If they don't get to extend this enum, what are they supposed to do? I think what's called for here is something like `codec::CodecError`'s `Other(#[source] Box<dyn Error + 'static + Send + Sync>)` variant.

That being said, this problem already existed (`VdafError::Flp` is arguably specific to the `Prio3` family), so I don't think it should be fixed in this PR: #424.

Reply:
@divergentdave seems to be following a design pattern already in this crate: Each module has its own error type. That said I agree. And the error cases are generic enough that I think we could just return `VdafError` instead of creating our own error type.

Reply:
I think it's correct for `mod Idpf` to define `IdpfError`, but it's not correct for that to appear as a variant in `VdafError`, which should be generic and not contain any VDAF's specifics.

Reply:
OOoh sorry I misunderstood.